http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -193,16 +190,17 @@ class HttpSamDAO[F[_]: Effect](httpClient: Client[F], config: HttpSamDaoConfig,
       )
     )(onError)
 
-  def getUserProxy(userEmail: WorkbenchEmail)(implicit ev: ApplicativeAsk[F, TraceId]): F[Option[WorkbenchEmail]] = {
-    val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, getAccessTokenUsingLeoPem))
-    httpClient.expectOptionOr[WorkbenchEmail](
-      Request[F](
-        method = Method.GET,
-        uri = config.samUri.withPath(s"/api/google/v1/user/proxyGroup/${userEmail.value}"),
-        headers = Headers.of(authHeader)
-      )
-    )(onError)
-  }
+  def getUserProxy(userEmail: WorkbenchEmail)(implicit ev: ApplicativeAsk[F, TraceId]): F[Option[WorkbenchEmail]] =
+    getAccessTokenUsingLeoJson.use { leoToken =>
```

Comment:
just to double check, it is intended that we'll be retrieving a new access token every time `getUserProxy` is used?

Reply:
Yeah I think this is intended.. we cache user pet tokens because that involves a call to Sam but not Leo SA tokens because we can generate them directly from the Leo SA json (which seems pretty performant).